src/ducks/apps/components/ApplicationPage/Details.jsx

```diff
@@ -40,20 +40,21 @@ export class Details extends Component {
     this.props.history.push(`/${parent}/${app.slug}/permissions`)
   }
 
-  toggleChannels() {
+  async toggleChannels() {
     const { app } = this.props
-    if (!getChannel(app.source))
-      console.warn("This application don't use the registry")
+    // only for installed apps and apps from the registry
+    if (!getChannel(app.source) || !app.installed) return
+    const stateUpdate = {}
     if (!this.state.displayBetaChannel) {
-      this.setState(() => ({ displayBetaChannel: true }))
+      stateUpdate.displayBetaChannel = true
     }
     if (!this.state.displayDevChannel) {
-      getContext().then(context => {
-        if (context && context.attributes && context.attributes.debug) {
-          this.setState(() => ({ displayDevChannel: true }))
-        }
-      })
+      const context = await getContext()
```

Comment:
I think we should not get context asynchronously every time we toggle a channel, but load it at application start. Then we could store it:
* in redux store
* in a CozyProvider and pass it to every children component of the application

Moreover, performing an async call on user event can make the whole UX feel slow, especially on mobile and even for a "developer" option.

The whole toggleChannels could be simplified as well:

```jsx
toggleChannels() {
   const { app } = this.props
   const hasAvailableChannels = app.installed && getChannel(app.source)
   if (!hasAvailableChannels) return

   const { showChannels } = this.state
   this.setState({
     showChannels: !showChannels
   })
})
```

and context could be used directly in the render method:

```jsx
  const displayDevChannel = !!this.context.debug
```

Channel beta should be always visible once `this.state.showChannels` is true.

Reply:
Agree about this change but this is not relevant for this PR since in any cases the context is fetched once then stored in a variable. I would like to have this fetch directly from the client and get this information from the Cozy Provider in the future.